code/modules/research/weaponsanalyzer.dm

```diff
@@ -252,9 +255,121 @@
+			dat += "Stuns: [initial(P.stun) ? "true" : "false"]<br>"
+			if(initial(P.shrapnel_type))
+				var/obj/shrapnel = new P.shrapnel_type
+				dat += "Shrapnel Type: [shrapnel.name]<br>"
+			dat += "Armor Penetration: [initial(P.armor_penetration)]%<br>"
+
+			if(istype(gun, /obj/item/gun/energy/laser/prototype))
+				var/obj/item/gun/energy/laser/prototype/E_prototype = gun
+				for(var/i in list(E_prototype.capacitor, E_prototype.focusing_lens, E_prototype.modulator) + E_prototype.gun_mods)
+					var/obj/item/laser_components/l_component = i
+
+					dat += "<br>Component Name: [initial(l_component.name)]</br><br>"
+					var/l_repair_name = initial(l_component.repair_item.name) ? initial(l_component.repair_item.name) : "nothing"
+					dat += "Reliability: [initial(l_component.reliability)]<br>"
+					dat += "Damage Modifier: [initial(l_component.damage)]<br>"
+					dat += "Fire Delay Modifier: [initial(l_component.fire_delay)]<br>"
+					dat += "Shots Modifier: [initial(l_component.fire_delay)]<br>"
+					dat += "Burst Modifier: [initial(l_component.burst)]<br>"
+					dat += "Accuracy Modifier: [initial(l_component.accuracy)]<br>"
+					dat += "Repair Tool: [l_repair_name]<br>"
+
+			if(E.secondary_projectile_type)
+				var/obj/item/projectile/P_second = E.secondary_projectile_type
+				dat += "<br><b>Secondary Projectile</b><br>"
+				dat += "Damage: [initial(P_second.damage)]<br>"
+				dat += "Damage Type: [initial(P_second.damage_type)]<br>"
+				dat += "Blocked by Armor Type: [initial(P_second.check_armor)]<br>"
+				dat += "Stuns: [initial(P_second.stun) ? "true" : "false"]<br>"
+				if(initial(P_second.shrapnel_type))
+					var/obj/second_shrapnel = new P_second.shrapnel_type
+					dat += "Shrapnel Type: [second_shrapnel.name]<br>"
+				dat += "Armor Penetration: [initial(P_second.armor_penetration)]%<br>"
+
+		else
+			var/obj/item/gun/projectile/P_gun = gun
+			var/obj/item/ammo_casing/casing = P_gun.ammo_type
+			var/obj/item/projectile/P = casing.projectile_type
+			dat += "Max Shots: [P_gun.max_shells]<br>"
+			dat += "Damage: [initial(P.damage)]<br>"
+			dat += "Damage Type: [initial(P.damage_type)]<br>"
+			dat += "Blocked by Armor Type: [initial(P.check_armor)]<br>"
+			dat += "Stuns: [initial(P.stun) ? "true" : "false"]<br>"
+			dat += "Shrapnel Type: [initial(P.shrapnel_type) ? initial(P.shrapnel_type) : "none"]<br>"
+		dat += "Burst: [gun.burst]<br>"
+		dat += "Reliability: [gun.reliability]<br>"
+
+	else
+		dat += "Damage: [item.force]<br>"
+		dat += "Damage Type: [item.damtype]<br>"
+		dat += "Sharp: [item.sharp ? "yes" : "no"]<br>"
+		dat += "Dismemberment: [item.edge ? "likely to dismember" : "unlikely to dismember"]<br>"
+		dat += "Penetration: [item.armor_penetration]<br>"
+		dat += "Throw Force: [item.throwforce]<br>"
+		if(istype(item, /obj/item/melee/energy))
+			var/obj/item/melee/energy/E_item = item
+			dat += "Active Damage: [item.force]<br>"
+			dat += "Active Throw Force: [item.force]<br>"
```

Comment:
This looks wrong. Throw force is its own var and I'm sure that active damage is not base force.